examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,291 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import AwsIcon from './aws-icon';
+import {Provider} from 'kepler.gl/cloud-providers';
+import window from 'global';
+import 'aws-sdk';
+import {Auth, Storage} from 'aws-amplify';
+import {MAP_URI} from '../../constants/default-settings';
+import {AWS_LOGIN_URL, AWS_WEB_CLIENT_ID} from './aws-login';
+
+export const PROVIDER_NAME = 'aws';
+export const DISPLAY_NAME = 'AWS';
+export const PRIVATE_STORAGE_ENABLED = true;
+export const SHARING_ENABLED = true;
+
+Storage.configure({level: 'private'});
+
+export default class AwsProvider extends Provider {
+  constructor() {
+    super({name: PROVIDER_NAME, displayName: DISPLAY_NAME, icon: AwsIcon});
+  }
+
+  /**
+   * Required!
+   * Idea / Hack: https://github.com/amazon-archives/amazon-cognito-identity-js/issues/508
+   *
+   * @param onCloudLoginSuccess
+   */
+  async login(onCloudLoginSuccess) {
+    const link = `${window.location.protocol}//${window.location.host}/${AWS_LOGIN_URL}`;
+    const style = `location, toolbar, resizable, scrollbars, status, width=500, height=440, top=200, left=400`;
+    const authWindow = window.open(link, 'awsCognito', style);
+
+    const handleLogin = async (e) => {
+      if (authWindow) {
+        authWindow.close();
+      }
+
+      window.removeEventListener('message', handleLogin);
+
+      if (e.data.success) {
+        onCloudLoginSuccess();
+      }
+    };
+    window.addEventListener('message', handleLogin);
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {Array<Viz>}
+   */
+  async listMaps() {
+    let s3PrivateFiles = await this._getMapListFromStorage('private');
+    let s3PublicFiles = await this._getMapListFromStorage('public');
+
+    return [...s3PublicFiles, ...s3PrivateFiles];
+  }
+
+  async _getMapListFromStorage(level) {
+    let mapList = [];
+    let capitalizedLevel = level.charAt(0).toUpperCase() + level.slice(1);
+    await Storage.list('', {level})
+      .then((result) => {
+        mapList = AwsProvider._updateFileList(result, level);
+      })
+      .catch((e) => {
+        this._handleError(`${capitalizedLevel} map load failed`, e);
+      });
+    return mapList;
+  }
+
+  static async _updateFileList(fileList, level) {
+    let updatedFileList = [];
+
+    for (const file of fileList) {
+      if (!file.key.endsWith('.json')) {
+        continue;
+      }
+      const title = file.key.split('.')[0];
+      const thumbnailKey = title + '.png';
+      const thumbnailURL = await Storage.get(thumbnailKey, {
+        level: level,
+        download: false
+      });
+
+      const description = await Storage.get(thumbnailKey, {
```

Comment:
Same comment as in line https://github.com/deepatlas/kepler.gl/pull/1/files#diff-9cc6944f72643588ae105aaffcf246beR81